File: airbyte-commons-worker/src/main/java/io/airbyte/workers/internal/AirbyteMessageBufferedWriter.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.workers.internal;

import io.airbyte.protocol.models.AirbyteMessage;
import java.io.IOException;

public interface AirbyteMessageBufferedWriter {

  void write(AirbyteMessage message) throws IOException;

  void flush() throws IOException;

  void close() throws IOException;

}

File: airbyte-cdk/java/airbyte-cdk/dependencies/bin/testFixtures/io/airbyte/workers/internal/AirbyteMessageBufferedWriter.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.workers.internal

import io.airbyte.protocol.models.AirbyteMessage
import java.io.IOException

interface AirbyteMessageBufferedWriter {
    @Throws(IOException::class) fun write(message: AirbyteMessage)

    @Throws(IOException::class) fun flush()

    @Throws(IOException::class) fun close()
}
